File: 204.count-primes.java
public class Solution {
    public int countPrimes(int n) {
        int [] a = new int [n+1];
        int [] ans = new int [n+1];

        boolean [] prime = new boolean [n+1];
        for (int i = 2; i < n; i++) {
            prime[i] = true;
        }

        int count = 0;

        for (int i = 2; i < n; i++) {
            if (prime[i] == true) {
                ans[count++] = i;
                for (int j = 2*i; j < n; j+=i) {
                    prime[j] = false;
                }
            }
        }


        return count;
    }
}

File: 204.count-primes.cpp
// CreateTime: 2020-12-03 15:32:01
class Solution {
public:
    int countPrimes(int n) {
        vector<bool> isPrime(n+2, true);

        isPrime[0] = false;
        isPrime[1] = false;

        int ans = 0;
        for (int i = 2; i <= n-1; i++) {
            if (isPrime[i]) {
                for (int j = 2; j <= (n-1)/i; j++) {
                    isPrime[i*j] = false;
                }
                ans++;
            }
        }
        return ans;
    }
};
